lighthouse-core/gather/drivers/driver.js

```diff
@@ -210,6 +211,42 @@ class Driver {
   }
 
   /**
+   * Checks all serviceworkes and see if any duplications are running
+   * @param {string} pageUrl
+   */
+  checkForMultipleServiceWorkers(pageUrl) {
+    // Get necessary information of serviceWorkers
+    const getRegistrations = new Promise(resolve => {
+      this.once('ServiceWorker.workerRegistrationUpdated', resolve);
+    });
+    const getVersions = this.getServiceWorkerVersions();
+
+    return Promise.all([getRegistrations, getVersions]).then(res => {
+      const registrations = res[0].registrations;
+      const versions = res[1].versions;
+
+      let activeServiceWorkers = [];
+      registrations
+        .filter(reg => !reg.isDeleted)
+        .filter(reg => parseURL(reg.scopeURL).hostname === parseURL(pageUrl).hostname)
+        .forEach(reg => {
+          activeServiceWorkers = activeServiceWorkers.concat(versions
+             .filter(ver => ver.registrationId === reg.registrationId)
+          );
+        });
+
+      if (activeServiceWorkers.length > 1) {
```

Comment:
I think we want to do something slightly different.

So far we have all worker versions that are attached to the registration that matches on the scopeURL. Perf. That's what we need... so far.

Our versions look like this: https://chromedevtools.github.io/debugger-protocol-viewer/tot/ServiceWorker/#type-ServiceWorkerVersion

We need all controlledClients of all `activeServiceWorkers`

if `totalControlledClients` has > 1 items, then we definitely have multiple tabs currently attached to the SW. if it's just one, then it is probably the open tab that's about to be reloaded.